db.h

```diff
@@ -102,7 +103,11 @@ class CAddrInfo {
   
   bool IsGood() const {
     if (ip.GetPort() != GetDefaultPort()) return false;
-    if (!(services & NODE_NETWORK)) return false;
+    if (clientVersion < NO_BLOOM_VERSION) {
+        if (!(services & NODE_NETWORK)) return false;
+    } else {
+        if (!(services & (NODE_NETWORK | NODE_BLOOM))) return false;
```

Comment:
I think you want: (services & (NODE_NETWORK | NODE_BLOOM)) != (NODE_NETWORK | NODE_BLOOM).